sdk/core/azure-core/tests/test_messaging_cloud_event.py

```diff
@@ -146,6 +144,108 @@ def test_cloud_custom_dict_with_extensions():
+    assert event.__class__ == CloudEvent
+    assert event.time.month == 2
+    assert event.time.day == 18
+    assert event.time.hour == 20
+    assert event.time.microsecond == 123000
+
+def test_cloud_custom_dict_ms_precision_is_eq_six():
+    cloud_custom_dict_with_extensions = {
+        "id":"de0fd76c-4ef4-4dfb-ab3a-8f24a307e033",
+        "source":"https://egtest.dev/cloudcustomevent",
+        "data":{"team": "event grid squad"},
+        "type":"Azure.Sdk.Sample",
+        "time":"2021-02-18T20:18:10.123456+00:00",
+        "specversion":"1.0",
+    }
+    event = CloudEvent.from_dict(cloud_custom_dict_with_extensions)
+    assert event.data == {"team": "event grid squad"}
+    assert event.__class__ == CloudEvent
+    assert event.time.month == 2
+    assert event.time.day == 18
+    assert event.time.hour == 20
+    assert event.time.microsecond == 123456
+
+def test_cloud_custom_dict_ms_precision_is_gt_six_z_not():
+    cloud_custom_dict_with_extensions = {
+        "id":"de0fd76c-4ef4-4dfb-ab3a-8f24a307e033",
+        "source":"https://egtest.dev/cloudcustomevent",
+        "data":{"team": "event grid squad"},
+        "type":"Azure.Sdk.Sample",
+        "time":"2021-02-18T20:18:10.539861122Z",
+        "specversion":"1.0",
+    }
+    event = CloudEvent.from_dict(cloud_custom_dict_with_extensions)
+    assert event.data == {"team": "event grid squad"}
+    assert event.__class__ == CloudEvent
+    assert event.time.month == 2
+    assert event.time.day == 18
+    assert event.time.hour == 20
+    assert event.time.microsecond == 539861
+
+def test_cloud_custom_dict_ms_precision_is_lt_six_z_not():
+    cloud_custom_dict_with_extensions = {
+        "id":"de0fd76c-4ef4-4dfb-ab3a-8f24a307e033",
+        "source":"https://egtest.dev/cloudcustomevent",
+        "data":{"team": "event grid squad"},
+        "type":"Azure.Sdk.Sample",
+        "time":"2021-02-18T20:18:10.123Z",
+        "specversion":"1.0",
+    }
+    event = CloudEvent.from_dict(cloud_custom_dict_with_extensions)
+    assert event.data == {"team": "event grid squad"}
+    assert event.__class__ == CloudEvent
+    assert event.time.month == 2
+    assert event.time.day == 18
+    assert event.time.hour == 20
+    assert event.time.microsecond == 123000
+
+def test_cloud_custom_dict_ms_precision_is_eq_six_z_not():
+    cloud_custom_dict_with_extensions = {
+        "id":"de0fd76c-4ef4-4dfb-ab3a-8f24a307e034",
+        "source":"https://egtest.dev/cloudcustomevent",
+        "data":{"team": "event grid squad"},
+        "type":"Azure.Sdk.Sample",
+        "time":"2021-02-18T20:18:10.123456Z",
+        "specversion":"1.0",
+    }
+    event = CloudEvent.from_dict(cloud_custom_dict_with_extensions)
+    assert event.data == {"team": "event grid squad"}
+    assert event.__class__ == CloudEvent
+    assert event.time.month == 2
+    assert event.time.day == 18
+    assert event.time.hour == 20
+    assert event.time.microsecond == 123456
```

Comment:
so is it that, no matter what, we can only check 6 digits? and the precision part comes when we're sending the event?

Reply:
Yes - no matter what, the decimal will have 6 digits